Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM golang:1.10.3-alpine3.8 AS builder
+
+RUN apk upgrade --update \
```

Comment:
separate these to 3 different command, so docker cache works better.
1. apk add git
2. apk upgrade --update
3. go get github.com/riobard/go-shadowsocks2

Reply:
This makes it possible to use the cache more efficiently, but the package versions upgraded by the `upgrade` command at different points in time are different;

If a package has a vulnerability and it is cached by docker, it may never be updated.

For the official docker image I think we should always keep the latest software version.